youtube/yt_data_extract/common.py

```diff
@@ -330,46 +330,29 @@ def extract_item_info(item, additional_info={}):
-        if 'index' in item: # url has wrong index on playlist page
-            info['index'] = extract_int(item.get('index'))
-        elif 'indexText' in item:
-            # Current item in playlist has ▶ instead of the actual index, must
-            # dig into url
-            match = re.search(r'index=(\d+)', deep_get(item,
-                'navigationEndpoint', 'commandMetadata', 'webCommandMetadata',
-                'url', default=''))
-            if match is None:   # worth a try then
-                info['index'] = extract_int(item.get('indexText'))
-            else:
-                info['index'] = int(match.group(1))
-        else:
-            info['index'] = None
-
-    elif primary_type == 'short':
-        info['id'] = item.get('videoId')
-        if not info['id']:
-            info['id'] = deep_get(item,'navigationEndpoint',
-                                  'reelWatchEndpoint', 'videoId')
-        info['approx_view_count'] = extract_approx_int(item.get('viewCountText'))
-
-        # handle case where it is "No views"
-        if not info['approx_view_count']:
-            if ('No views' in extract_str(item.get('viewCountText', ''))):
-                info['view_count'] = 0
-                info['approx_view_count'] = '0'
-
-        # dig into accessibility data to get duration for shorts
-        accessibility_label = multi_deep_get(item,
-            ['accessibility', 'accessibilityData', 'label'],
-            default='')
-        duration = re.search(r'(\d+) (second|seconds|minute) - play video$',
-                             accessibility_label)
-        if duration:
-            if duration.group(2) == 'minute':
-                info['duration'] = '1:00'
-            else:
-                info['duration'] = '0:' + duration.group(1).zfill(2)
+        if info['duration'] is None: # shorts
+            if not info['id']:
+                info['id'] = deep_get(item,'navigationEndpoint',
+                                    'reelWatchEndpoint', 'videoId')
+            info['approx_view_count'] = extract_approx_int(item.get('viewCountText'))
+
+            # handle case where it is "No views"
+            if not info['approx_view_count']:
+                if ('No views' in extract_str(item.get('viewCountText', ''))):
+                    info['view_count'] = 0
+                    info['approx_view_count'] = '0'
+
+            # dig into accessibility data to get duration for shorts
+            accessibility_label = multi_deep_get(item,
+                ['accessibility', 'accessibilityData', 'label'],
+                default='')
+            duration = re.search(r'(\d+) (second|seconds|minute) - play video$',
+                                accessibility_label)
+            if duration:
+                if duration.group(2) == 'minute':
+                    info['duration'] = '1:00'
+                else:
+                    info['duration'] = '0:' + duration.group(1).zfill(2)
```

Comment:
I had more in mind including only the duration stuff in this kind of separate check. If the video id, viewcounttext, and accessibility label have different keys, those can go as separate possibilities next to the ones for regular videos. For example, further up in the file you'll see
``` 
    info['thumbnail'] = normalize_url(multi_deep_get(item,
        ['thumbnail', 'thumbnails', 0, 'url'],      # videos
        ['thumbnails', 0, 'thumbnails', 0, 'url'],  # playlists
        ['thumbnailRenderer', 'showCustomThumbnailRenderer', 'thumbnail', 'thumbnails', 0, 'url'], # shows
    ))
```
That's the idea. For video_id, you replace
```
   if primary_type == 'video':
        info['id'] = item.get('videoId')
        if not info['id']:
            info['id'] = deep_get(item,'navigationEndpoint', 'watchEndpoint',
                                  'videoId')
```
with something like
```
    info['id'] = multi_deep_get(item,
                       ['videoId'],
                       ['navigationEndpoint', 'watchEndpoint', 'videoId']
                       ['navigationEndpoint', 'reelWatchEndpoint', 'videoId'] # shorts
    )
```

It appears that 'viewCountText' for regular videos contains all the digits and 'shortViewCountText' does not and has e.g. 5K views, whereas for shorts they will put '5K views' into viewCountText. Not sure what's going on there but handle it in a way that makes sense, perhaps you try running approx_extract_int on viewCountText if no view counts have been successfully extracted.

For duration from the accessibility label, you can also use the conservative_update function if you'd like, whatever matches the surrounding code better

Reply:
I see. Sorry about that.
Changes made. I opted to remove the `if info['duration'] is None` check in favor of conservative_update for consistency with the recommended videos' view_count and time_published extraction process. I also simply used multi_get in the path where view_count is missing to extract approx_view_count for shorts. Hopefully this looks cleaner than the previous attempts.